chainer/functions/pooling/average_pooling_2d.py

```diff
@@ -65,8 +66,7 @@ def backward_cpu(self, x, gy):
         return gx,
 
     def backward_gpu(self, x, gy):
-        if (cuda.cudnn_enabled and self.use_cudnn and
-                pooling_2d._check_cudnn_acceptable_type(x[0].dtype)):
+        if self._used_cudnn:
```

Comment:
Yes, that's right. cuDNN routine requires `self.y` in backward, which is not required and thus retained by CuPy routine, and so we cannot use cuDNN only for backward unless we retain `self.y` in CuPy routine as well.